SwiftGit2/Repository.swift

```diff
@@ -537,4 +537,297 @@ final public class Repository {
+		guard lookupParentGitResult == GIT_OK.rawValue, let unwrapParentCommit = unsafeParentCommit else {
+			return Result.failure(NSError(gitError: lookupParentGitResult, pointOfFailure: "git_commit_lookup"))
+		}
+		git_commit_free(unsafeParentCommit)
+
+		var unsafeParentTree: OpaquePointer? = nil
+		let parentTreeResult = git_commit_tree(&unsafeParentTree, unwrapParentCommit)
+		guard parentTreeResult == GIT_OK.rawValue, let unwrapParentTree = unsafeParentTree else {
+			return Result.failure(NSError(gitError: parentTreeResult, pointOfFailure: "git_commit_tree"))
+		}
+		git_tree_free(unsafeParentTree)
+
+		var unsafeDiff: OpaquePointer? = nil
+		let diffResult = git_diff_tree_to_tree(&unsafeDiff, self.pointer, unwrapParentTree, baseTree, nil)
+		guard diffResult == GIT_OK.rawValue, let unwrapDiffResult = unsafeDiff else {
+			return Result.failure(NSError(gitError: diffResult, pointOfFailure: "git_diff_tree_to_tree"))
+		}
+
+		return self.processDiffDeltas(unwrapDiffResult)
+	}
+
+	private func getDiffDeltasWithMultipleParents(from baseTree: OpaquePointer,
+	                                              in commit: Commit) -> Result<[GitDiffDelta], NSError> {
+		// Merge Commit, merge diffs of base with each parent
+		var mergeDiff: OpaquePointer? = nil
+		for parent in commit.parents {
+			var unsafeParentCommit: OpaquePointer? = nil
+			let unsafeParentOid = UnsafeMutablePointer<git_oid>.allocate(capacity: 1)
+			git_oid_fromstr(unsafeParentOid, parent.oid.description)
+			let lookupParentGitResult = git_commit_lookup(&unsafeParentCommit, self.pointer, unsafeParentOid)
+			guard lookupParentGitResult == GIT_OK.rawValue, let unwrapParentCommit = unsafeParentCommit else {
+				return Result.failure(NSError(gitError: lookupParentGitResult, pointOfFailure: "git_commit_lookup"))
+			}
+			git_commit_free(unsafeParentCommit)
+
+			var unsafeParentTree: OpaquePointer? = nil
+			let parentTreeResult = git_commit_tree(&unsafeParentTree, unwrapParentCommit)
+			guard parentTreeResult == GIT_OK.rawValue, let unwrapParentTree = unsafeParentTree else {
+				return Result.failure(NSError(gitError: parentTreeResult, pointOfFailure: "git_commit_tree"))
+			}
+			git_tree_free(unsafeParentTree)
+
+			var unsafeDiff: OpaquePointer? = nil
+			let diffResult = git_diff_tree_to_tree(&unsafeDiff, self.pointer, unwrapParentTree, baseTree, nil)
+			guard diffResult == GIT_OK.rawValue, let unwrapDiffResult = unsafeDiff else {
+				return Result.failure(NSError(gitError: diffResult, pointOfFailure: "git_diff_tree_to_tree"))
+			}
+
+			if mergeDiff == nil {
+				mergeDiff = unwrapDiffResult
+			} else {
+				let mergeResult = git_diff_merge(mergeDiff, unwrapDiffResult)
+				guard mergeResult == GIT_OK.rawValue else {
+					return Result.failure(NSError(gitError: mergeResult, pointOfFailure: "git_diff_merge"))
+				}
+			}
+		}
+		return self.processDiffDeltas(mergeDiff!)
```

Comment:
This code doesn't look any different than the code in `getDiffDeltasWithOneParent`, so I think that function can be removed. This function can be used instead and the loop will do the right thing.